include/upgrader/streams/core/e7038ce9-ddbe2e76.task.php

```diff
@@ -87,6 +87,30 @@ function run() {
                  .', created=NOW(), updated=NOW(), isactive=1';
              db_query($sql);
          }
+
+         // See if there are missing events that should be added to the database
+         $event_type = array('login', 'logout', 'message', 'note');
+         foreach($event_type as $eType) {
+             $sql = sprintf("SELECT * FROM `%s` WHERE name = '%s'",
+                     TABLE_PREFIX.'event', $eType);
+
+             $res=db_query($sql);
+
+             if($count = db_num_rows($res) && $count > 0)
+                return $this->error('Event already exists');
```

Comment:
Are you sure you want to error out (return) here? I would think continue will be sufficient.